test/test.go

```diff
@@ -80,10 +80,11 @@ type TestCase struct {
 
 	MultitenancySettings   *MultitenancySettings
 	DisableTenantResources bool
+	StrictlyTenantScoped   bool
```

Comment:
I'm not sure if we can completely get rid of this flag as I see there are some tests that are some generic tests executed only if StrictlyTenantScoped is true.

But there were numerous places where we do the exact same request just a different SMExpect (either SMWithOAuth, or SMWithAuthForTenant) - what do you think about extending the TestCase struct to contain also the SMExpect with which the tests should be executed? This way we can remove some of those if/else checks. Not sure if this can help somehow with the StrictlyTenantScoped flag.